fs/ftpfs.py

```diff
@@ -99,16 +107,19 @@ def __init__(self, ftpfs, path, mode):
         self._read_conn = None
         self._write_conn = None
 
+    def __length_hint__(self):
+        return self.fs.getsize(self.path)
+
     def _open_ftp(self):
-        ftp = self.fs._open_ftp()
-        ftp.voidcmd(_encode('TYPE I'))
+        ftp = self.fs._open_ftp(self.fs.encoding)
+        ftp.voidcmd(str('TYPE I'))
```

Comment:
So `str` would be bytes on Python 2.7 and text type on Py3? Which is what the ftp library expects?

Reply:
So, this is a summary from what I read during my researches:

`ftplib` expected binary (`str`) strings in Python 2, but was patched to simply decode text (`str`) strings in Python 3 and decode them as UTF-8. Because of this, there may be cases where a string is decode twice, or things like that. I'm not completely sure about it (the StackOverflow where people debated that stated the `ftplib` was a mess), but at least that provided good results with unicode path tests : when I did it another way, I'd start to have decoding/encoding errors in either Python 2 or Python 3.